ecosystem/typescript/sdk_v2/src/core/account.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import nacl from "tweetnacl";
+import * as bip39 from "@scure/bip39";
+import { AccountAddress } from "./account_address";
+import { Hex } from "./hex";
+import { bytesToHex } from "@noble/hashes/utils";
+import { HexInput } from "../types";
+import { PrivateKey, PublicKey, Signature } from "../crypto/ed25519";
+import { derivePath } from "../utils/hd-key";
+import { AuthenticationKey } from "../crypto/authentication_key";
+
+/**
+ * Class for creating and managing account on Aptos network
+ *
+ * Use this class to create accounts, sign transactions, and more.
+ * Note: Creating an account instance does not create the account onchain.
+ */
+export class Account {
+  /**
+   * A private key and public key, associated with the given account
+   */
+  readonly publicKey: PublicKey;
+  readonly privateKey: PrivateKey;
+
+  /**
+   * Account address associated with the account
+   */
+  readonly accountAddress: AccountAddress;
+
+  /**
+   * constructor for Account
+   *
+   * @param args.privateKey PrivateKey - private key of the account
+   * @param args.address AccountAddress - address of the account
+   *
+   * This method is private because it should only be called by the factory static methods.
+   * @returns Account
+   */
+  private constructor(args: { privateKey: PrivateKey; address: AccountAddress }) {
+    const { privateKey, address } = args;
+
+    // Derive the public key from the private key
+    const keyPair = nacl.sign.keyPair.fromSeed(privateKey.toUint8Array().slice(0, 32));
+    this.publicKey = new PublicKey({ hexInput: keyPair.publicKey });
+
+    this.privateKey = privateKey;
+    this.accountAddress = address;
+  }
```

Comment:
This is tied to an Ed25519 key, it should probably take in some interface supporting multiple types of keys.  Otherwise, it doesn't support MultiEd25519, secpk etc.

Reply:
Yeah I see there is an active new scheme `secp256k1` being added now. Will try to sync up more closely with what @davidiw had on `asymmetric_keys` - https://github.com/aptos-labs/aptos-core/pull/10200

Reply:
Updated to args to be an abstract class of `PrivateKey`